pennylane/_device.py

```diff
@@ -705,11 +705,6 @@ def batch_transform(self, circuit):
             the sequence of circuits to be executed, and a post-processing function
             to be applied to the list of evaluated circuit results.
         """
-
-        # If the observable contains a Hamiltonian and the device does not
-        # support Hamiltonians, or if the simulation uses finite shots, or
-        # if the Hamiltonian explicitly specifies an observable grouping,
-        # split tape into multiple tapes of diagonalizable known observables.
```

Comment:
This might cause some merge conflicts with #2587 (@Qottmann), something to be aware of for whoever merges second 😆